ecl/eclcmd/eclcmd_core.cpp

```diff
@@ -1703,6 +1703,126 @@ class EclCmdStatus : public EclCmdCommon
+            }
+            if (iter.matchOption(optPath, ECLOPT_PATH))
+            {
+                retVal = EclCmdOptionMatch;
+                continue;
+            }
+            if (iter.matchOption(optIncThorSlave, ECLOPT_INC_THOR_SLAVE_LOGS))
+            {
+                retVal = EclCmdOptionMatch;
+                continue;
+            }
+            if (iter.matchOption(optProblemDesc, ECLOPT_PROBLEM_DESC))
+            {
+                retVal = EclCmdOptionMatch;
+                continue;
+            }
+            eclCmdOptionMatchIndicator ind = EclCmdCommon::matchCommandLineOption(iter, true);
+            if (ind != EclCmdOptionMatch)
+                return ind;
+        }
+        return retVal;
+    }
+    virtual bool finalizeOptions(IProperties *globals)
+    {
+        if (!EclCmdCommon::finalizeOptions(globals))
+            return false;
+        return true;
+    }
+    virtual int processCMD()
+    {
+        if (optWuid.isEmpty())
+        {
+            fprintf(stdout, "No WUID .\n");
+            return 0;
+        }
+        Owned<IClientWsWorkunits> client = createCmdClient(WsWorkunits, *this);
+
+        Owned<IClientWUCreateZAPInfoRequest> zapReq = client->createWUCreateZAPInfoRequest();
+        zapReq->setWuid(optWuid.get());
+        zapReq->setIncludeThorSlaveLog(optIncThorSlave);
+        zapReq->setProblemDescription(optProblemDesc.get());
+
+        Owned<IClientWUCreateZAPInfoResponse> zapResp = client->WUCreateZAPInfo(zapReq);
+        int ret = outputMultiExceptionsEx(zapResp->getExceptions());
+        if (ret == 0)
+        {
+            StringBuffer filePath(optPath);
+            const char *p = filePath.str();
+            p = (p + strlen(p) - 1);
+            if (!streq(p, PATHSEPSTR))
+                filePath.append(PATHSEPSTR);
+
+            filePath.append(zapResp->getZAPFileName());
+
+            int ofile = open(filePath.str(), _O_WRONLY | _O_CREAT | _O_TRUNC, _S_IREAD | _S_IWRITE);
+            if(ofile == -1)
+                throw MakeStringException(-1, "can't open file %s, ZAP file creation cancelled.\n", filePath.str());
+
+            const MemoryBuffer & zapFile = zapResp->getThefile();
+            ssize_t written = write(ofile, zapFile.toByteArray(), zapFile.length());
+            if (written < 0)
+                throw MakeStringException(-1, "can't write to file %s, ZAP file creation cancelled.\n", filePath.str());
+
+            if (written != zapFile.length())
+                throw MakeStringException(-1, "truncated write to file %s, ZAP file creation cancelled.\n", filePath.str());
+
+            close(ofile);
+
+            fprintf(stdout, "ZAP file written into %s.\n",filePath.str());
+        }
+
+        return 0;
+    }
+    virtual void usage()
+    {
+        fputs("\nUsage:\n"
+            "\n"
+            "Create and store ZAP file of the given workunit.\n"
+            "\n"
+            "ecl status -wu <WUID> -path <zap_file_path>\n"
```

Comment:
Change "status" to "zapgen".

Reply:
If WUID is always required, you can make it a parameter instead of an option.  That is:

ecl zapgen W20170914-142053

instead of

ecl zapgen -wu W20170914-142053

Reply:
Example code for EclCmdResults:

const char *arg = iter.query();
            if (*arg!='-')  //parameters don't start with '-'
            {
                if (optWuid.length())
                {
                    fprintf(stderr, "\nunrecognized argument %s\n", arg);
                    return EclCmdOptionCompletion;
                }
                if (!looksLikeAWuid(arg, 'W'))
                {
                    fprintf(stderr, "\nargument should be a workunit id: %s\n", arg);
                    return EclCmdOptionCompletion;
                }
                optWuid.set(arg);
                continue;
            }

Reply:
Changed